src/nitypes/vector.py

```diff
@@ -0,0 +1,235 @@
+        return self._values[index]
+
+    @overload
+    def __setitem__(  # noqa: D105 - missing docstring in magic method
+        self, index: int, value: VectorType
+    ) -> None: ...
+
+    @overload
+    def __setitem__(  # noqa: D105 - missing docstring in magic method
+        self, index: slice, value: Iterable[VectorType]
+    ) -> None: ...
+
+    @override
+    def __setitem__(self, index: int | slice, value: VectorType | Iterable[VectorType]) -> None:
+        """Set value(s) at the specified location."""
+        if isinstance(index, int):
+            if not isinstance(value, self._value_type):
+                raise self._create_value_mismatch_exception(value)
+
+            if isinstance(value, int):
+                self._validate_int_value(value)
+
+            self._values[index] = value
+        else:  # slice
+            if not isinstance(value, Iterable):
+                raise TypeError("You must assign an Iterable to a Vector slice.")
+            elif isinstance(value, str):  # Narrow the type to exclude string.
+                raise TypeError("You cannot assign a string to Vector slice.")
+            else:
+                for subval in value:
+                    if not isinstance(subval, self._value_type):
+                        raise self._create_value_mismatch_exception(subval)
+                    elif isinstance(subval, int):
+                        self._validate_int_value(subval)
+
+            self._values[index] = value
+
+    def __delitem__(self, index: int | slice) -> None:
+        """Delete item(s) from the specified location."""
+        del self._values[index]
+
+    def __len__(self) -> int:
+        """Return the length of the Vector."""
+        return len(self._values)
+
+    def insert(self, index: int, value: VectorType) -> None:
+        """Insert a value at the specified location."""
+        if not isinstance(value, self._value_type):
+            raise self._create_value_mismatch_exception(value)
+
+        if isinstance(value, int):
+            self._validate_int_value(value)
+
+        self._values.insert(index, value)
+
+    def __eq__(self, value: object, /) -> bool:
+        """Return self==value."""
+        if not isinstance(value, self.__class__):
+            return NotImplemented
+        return self._values == value._values and self.units == value.units
+
+    def __reduce__(self) -> tuple[Any, ...]:
+        """Return object state for pickling."""
+        return (self.__class__, (self._values, self.units))
+
+    def __repr__(self) -> str:
+        """Return repr(self)."""
+        args = [f"values={self._values!r}", f"units={self.units!r}"]
+        return f"{self.__class__.__module__}.{self.__class__.__name__}({', '.join(args)})"
+
+    def __str__(self) -> str:
+        """Return str(self)."""
+        if self.units:
+            values_with_units = [f"{value} {self.units}" for value in self._values]
+            return ", ".join(values_with_units)
```

Comment:
This doesn't convey the fact that the value is a collection. Consider adding brackets.